In [1]:
# import libraries
import pandas as pd
from sqlalchemy import create_engine

In [2]:
#creating connection to the database
db_config = {'user': 'test',         # user name
             'pwd': 'test', # password
             'host': 'test',
             'port': 6432,              # connection port
             'db': 'test'}          # the name of the database

connection_string = 'postgresql://{}:{}@{}:{}/{}'.format(db_config['user'], db_config['pwd'], db_config['host'], db_config['port'], db_config['db'])

engine = create_engine(connection_string)

The database connection is stored in the engine variable. Let's run an SQL query:

In [3]:
query = '''SELECT * FROM trending_by_time'''
trending_by_time = pd.io.sql.read_sql(query, con = engine, index_col = 'record_id')

Let's download data from the trending_by_time table to the file trending_by_time.csv

In [ ]:
trending_by_time.to_csv('trending_by_time.csv', index = False)

### Studing the general information about table

In [ ]:
#openinging the table
df = pd.read_csv('trending_by_time.csv')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12343 entries, 0 to 12342
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   region          12343 non-null  object
 1   trending_date   12343 non-null  object
 2   category_title  12343 non-null  object
 3   videos_count    12343 non-null  int64 
dtypes: int64(1), object(3)
memory usage: 385.8+ KB


- there are no missing values

In [7]:
df.head()

region trending_date    category_title  videos_count
0  France    2017-11-14  Autos & Vehicles             8
1  France    2017-11-15  Autos & Vehicles             2
2  France    2017-11-16  Autos & Vehicles             6
3  France    2017-11-17  Autos & Vehicles             8
4  France    2017-11-18  Autos & Vehicles             4

We have 4 columns:  
- region - region 
- trending_date - date when videos in trends
- category_title - specific category of videos
- videos_count - number of videos in the trending section

In [8]:
df.region.value_counts()

Russia           2930
United States    2860
France           2774
India            2283
Japan            1496
Name: region, dtype: int64

- we have 5 regions

In [12]:
print('Start date:', df.trending_date.min())
print('End date:', df.trending_date.max())

Start date: 2017-11-14
End date: 2018-06-14


First date in our data is 2017-11-14, last date: 2018-06-14

In [23]:
print('Number of categories:', df.category_title.value_counts().count())
df.category_title.value_counts()

Number of categories: 18


People & Blogs           942
Entertainment            942
Comedy                   940
News & Politics          939
Film & Animation         938
Music                    938
Howto & Style            937
Sports                   933
Science & Technology     881
Education                863
Gaming                   767
Autos & Vehicles         681
Pets & Animals           674
Travel & Events          489
Shows                    406
Nonprofits & Activism     49
Movies                    22
Trailers                   2
Name: category_title, dtype: int64

In [14]:
df.videos_count.describe()

count    12343.000000
mean        27.545167
std         29.793491
min          2.000000
25%          8.000000
50%         18.000000
75%         36.000000
max        220.000000
Name: videos_count, dtype: float64

Number of videos range from 2 to 220 per date.  
Average number of videos per day is 27.

In [21]:
print('Total number of videos:', df.videos_count.sum())

Total number of videos: 339990


Let's create a dashboard.  
Link to dashboard: https://public.tableau.com/profile/sofiya6834#!/vizhome/Project_10_16065728622970/Dashboard?publish=yes

Link to presentation: https://docs.google.com/presentation/d/1kvj6hZn6V9X7Su9SqzLtFDLgDdOnGCIsz2B645lTojk/edit?usp=sharing